File: 07-UserSignupProcessManager.java
@Configuration
public class UserSignupProcessManager {

    @Bean
    public DomainEventListener<UserSignedEvent> onUserSigned(RequestUserEmailVerificationUsecase requestEmailConfirmationUsecase) {
        return DomainEventListener.create(UserSignedEvent.class, ee ->
            requestEmailConfirmationUsecase.execute(ee.getUserId()));
    }
}

File: 07-UserSignupProcessManager.kt
@Configuration(proxyBeanMethods = false)
class UserSignupProcessManager {
    @Bean
    fun onUserSigned(requestEmailConfirmationUsecase: RequestUserEmailVerificationUsecase) =
        domainEventListener<UserSignedEvent> {
            requestEmailConfirmationUsecase.execute(it.userId!!)
        }
}
